check.sh

```diff
@@ -8,139 +8,195 @@
+function run() {
+    # https://stackoverflow.com/a/76153233/14637
+    echo -n '>' >&2
+    for arg in "$@"; do
+        printf " %q" "$arg" >&2
+    done
+    echo >&2
+    "$@"
+}
 
-# For integration tests
+# Finds the Godot binary and stores its path in $godotBin. Logs an error and returns with nonzero
+# exit status if not found.
 function findGodot() {
-    # User-defined GODOT4_BIN
-    if [ -n "$GODOT4_BIN" ]; then
-        echo "Found GODOT4_BIN env var ($GODOT4_BIN)"
+    # $godotBin previously detected.
+    if [[ -v godotBin ]]; then
+        return
+
+    # User-defined GODOT4_BIN.
+    elif [[ -n "$GODOT4_BIN" ]]; then
+        log "Using environment variable GODOT4_BIN=$(printf %q "$GODOT4_BIN")"
         godotBin="$GODOT4_BIN"
 
-    #  Executable in path
-    elif command -v godot4 &>/dev/null; then
-        echo "Found 'godot4' executable"
+    # Executable in path.
+    elif command -v godot4 >/dev/null; then
+        log "Found 'godot4' executable"
         godotBin="godot4"
 
-    # Special case for Windows when there is a .bat file
+    # Special case for Windows when there is a .bat file.
     # Also consider that 'cmd /c' would need 'cmd //c' (https://stackoverflow.com/q/21357813)
     elif
-        godot4.bat --version
-        [[ $? -eq 0 ]]
+        godot4.bat --version 2>/dev/null
     then
-        echo "Found 'godot4.bat' script"
+        log "Found 'godot4.bat' script"
         godotBin="godot4.bat"
 
-    # This should come last: only use this as a last resort as usually `godot`
-    # refers to a Godot 3.x installation.
-    elif command -v godot &>/dev/null; then
+    # This should come last: only use this as a last resort as `godot` may refer to a 
+    # Godot 3.x installation.
+    elif command -v godot >/dev/null; then
         # Check if `godot` actually is Godot 4.x
-        if godot --version | grep -qE "^4\\."; then
-            echo "Found 'godot' executable with version $(godot --version)"
+        godotVersion="$(command godot --version)"
+        if [[ "$godotVersion" =~ ^4\. ]]; then
+            log "Found 'godot' executable with version $godotVersion"
             godotBin="godot"
         else
-            echo "Found 'godot' executable, but it has the incompatible version $(godot --version)"
-            exit 2
+            log "Found 'godot' executable, but it has incompatible version $godotVersion"
+            return 1
         fi
 
-    # Error case
+    # Error case.
     else
-        echo "Godot executable not found"
-        exit 2
+        log "Godot executable not found; try setting GODOT4_BIN to the full path to the executable"
+        return 1
     fi
 }
 
+################################################################################
+# Commands
+################################################################################
+
+# Poor human's namespacing: all commands are prefixed with `cmd_` to avoid confusion with shell
+# builtins like `test`.
```

Comment:
Nitpick, I think the idiom is called "poor man's" 🙂

Reply:
I like poor human 😆

Reply:
I think "man" in that context means human, as in "mankind". Might be wrong though. Just sounds a bit weird to change a known idiom, but I am also not a native speaker... 😉

Reply:
It's now gone anyway.